compiler/noirc_evaluator/src/ssa/opt/mem2reg.rs

```diff
@@ -274,6 +273,13 @@ impl<'f> PerFunctionContext<'f> {
                     self.instructions_to_remove.insert(*last_store);
                 }
 
+                if let Some(last_load) = self.last_loads.get(&address) {
+                    let load_result = self.inserter.function.dfg.instruction_results(*last_load)[0];
+                    if load_result == value {
+                        self.instructions_to_remove.insert(instruction);
+                    }
+                }
```

Comment:
I'm unsure on the correctness of this (and last_loads as a whole) in the presence of aliasing. If we had e.g.
```rs
let foo = &mut 0;
let alias = foo;

let foo_load1 = *foo;
*alias = 7;
*foo = foo_load1; // This looks like it'd be optimized out

assert_eq(*foo, 7);
```

Reply:
We expect this assert to fail right? This still fails with these changes. 

Either way this approach does look to have aliasing issues, specifically with arrays so I'm going to have to update this approach.